### PR TITLE
Add dev.vieb.Vieb

### PR DESCRIPTION
**Pull request by tinywrkb**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**  
  The developer of the app has no plans for Flathub submission, and I received approval for submitting it.
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

~~Note that the `--filesystem=home` permission was given, as it looks like the app does not respect `user-dirs.dirs`, so if the user has set a different value from the default `XDG_DOWNLOAD_DIR=$HOME/Downloads`, then the app won't download files to the correct folder that was mounted into the sandbox with `--filesystem=xdg-downloads`, or at least with the default shipped settings.~~

**CC** @Jelmerro <- should be added as collaborator.  
Keeping as draft until the app's developer will give the thumbs up here.

### CODE REVIEW

**Comment by tinywrkb** [comment]
bot, build dev.vieb.Vieb

File: dev.vieb.Vieb.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      <image>https://vieb.dev/img/2.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://vieb.dev/img/5.png</image>
+    </screenshot>
+  </screenshots>
+  <releases>
+    <release version="7.2.0" date="2022-04-01"/>
+  </releases>
+  <launchable type="desktop-id">dev.vieb.vieb.desktop</launchable>
```
Comment (hfiguiere) [inline dev.vieb.Vieb.metainfo.xml:48]:
```suggestion
  <launchable type="desktop-id">dev.vieb.Vieb.desktop</launchable>
```

Comment (tinywrkb) [reply dev.vieb.Vieb.metainfo.xml]: Thanks! Fixed.

**Comment by tinywrkb** [comment]
bot, build dev.vieb.Vieb

**Comment by tinywrkb** [comment]
I don't have an aarch64 platform to test on, and Electron apps segfault when running with qemu-user, even when disabling Chrome's sandbox and forcing software rendering in mesa with 
```
export LIBGL_ALWAYS_SOFTWARE=1
export __GLX_VENDOR_LIBRARY_NAME=mesa
export VK_ICD_FILENAMES=/usr/lib/aarch64-linux-gnu/GL/vulkan/icd.d/lvp_icd.aarch64.json
```
So idea if it works correctly on the architecture.

**Comment by hfiguiere** [comment]
works fine here on actual hardware. However, I get:

```
MESA-LOADER: failed to retrieve device information
MESA-LOADER: failed to retrieve device information
MESA-LOADER: failed to retrieve device information
MESA-LOADER: failed to open kms_swrast: /usr/lib/aarch64-linux-gnu/GL/default/lib/dri/kms_swrast_dri.so: cannot open shared object file: Permission denied (search paths /usr/lib/aarch64-linux-gnu/GL/default/lib/dri, suffix _dri)
failed to load driver: kms_swrast
MESA-LOADER: failed to open swrast: /usr/lib/aarch64-linux-gnu/GL/default/lib/dri/swrast_dri.so: cannot open shared object file: Permission denied (search paths /usr/lib/aarch64-linux-gnu/GL/default/lib/dri, suffix _dri)
failed to load swrast driver
```

**Comment by tinywrkb** [comment]
What about when running with 

`--env=LIBGL_DRIVERS_PATH=/usr/lib/aarch64-linux-gnu/GL/default/lib/dri:/usr/lib/aarch64-linux-gnu/GL/lib/dri`

**Comment by tinywrkb** [comment]
Also, what about other Electron apps built for aarch64? For example, `net.cozic.joplin_desktop`, though this one is built from source.

**Comment by Jelmerro** [comment]
Glad to see that this will get a flatpak, I had no plans for this as it's non-trivial to add it to the existing build process, so I'm glad that someone else is willing to make a package for it on flathub. Let me know if you need anything from me, besides my approval of this package.

**Comment by hfiguiere** [comment]
> Also, what about other Electron apps built for aarch64? For example, `net.cozic.joplin_desktop`, though this one is built from source.

dunno. just tested this one because you inquired whether it worked.

**Comment by tinywrkb** [comment]
> > Also, what about other Electron apps built for aarch64? For example, `net.cozic.joplin_desktop`, though this one is built from source.
> 
> dunno. just tested this one because you inquired whether it worked.

Alright, thanks! I'm not quite sure what's going on.  
`/usr/lib/aarch64-linux-gnu/GL/default` should be there, Flatpak does mount the `org.freedesktop.Platform.GL.default` extension on this extension mount point, at least I'm entering a sandbox using the `org.freedesktop.Platform/aarch64/21.08` runtime, and with the help of qemu-user. 

Anyway, as the application does run, I'm considering this as non-blocking and ready for merge.

**Comment by tinywrkb** [comment]
> it is there, I checked.

So maybe this is SoC platform specific issue, not architecture related, and an update to Electron or mesa might resolve this.  
It's looking for mesa's software rendering, so it's likely that using swiftshader doesn't change much for the user.  
Post merge, I will try switching this to build from source. Maybe that would help.

**Comment by tinywrkb** [comment]
bot, build dev.vieb.Vieb

**Comment by tinywrkb** [comment]
Building now from source. flathubbot f-e-d-c checks are disabled due to the obvious reasons.  
Only missing now an update workflow that will take care of running `flatpak-node-generator.py`. Will likely follow the Joplin example.

**Comment by tinywrkb** [comment]
bot, build dev.vieb.Vieb

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "automerge-flathubbot-prs": true,
```
[the remark this answers is not in the document]
Comment (tinywrkb) [reply flathub.json]: Fixed in d9915c0712094b35b3afe6f3cf7fe1b693dc3ea8

File: dev.vieb.Vieb.yaml [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node18
+command: vieb
+finish-args:
+  - --device=all
+  - --env=GTK_PATH=/app/lib/gtkmodules
+  - --filesystem=/run/.heim_org.h5l.kcm-socket
+  - --filesystem=home
```
[the remark this answers is not in the document]
Comment (tinywrkb) [reply dev.vieb.Vieb.yaml]: Fixed in 756bea100a8010b7b7cdb4b50f515cccf812d51c

**Comment by tinywrkb** [comment]
bot, build dev.vieb.Vieb

File: dev.vieb.Vieb.yaml [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  - --filesystem=xdg-run/pipewire-0:ro
+  - --own-name=org.mpris.MediaPlayer2.chromium.*
+  - --require-version=1.8.2
+  - --share=ipc
+  - --share=network
+  - --socket=cups
+  - --socket=pulseaudio
+  - --socket=wayland
+  - --socket=x11
```
[the remark this answers is not in the document]
Comment (barthalion) [reply dev.vieb.Vieb.yaml:27]: It should have `--socket=wayland` removed altogether then. If someone is happy to stray off the default path, they can also customize flatpak permissions.

[the remark this answers is not in the document]
Comment (tinywrkb) [reply dev.vieb.Vieb.yaml:26]: Basically all Chromium-based browser do the same, but only some of the Electron apps I've checked.  
Fixed in 61523d8ece2f3b0b29087f6126b39f6b982683db.

**Comment by tinywrkb** [comment]
bot, build dev.vieb.Vieb

**Pull request closed without merging**